core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -240,6 +252,10 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
                     return;
                 }
 
+                final ClientConnectionTimingsBuilder timingsBuilder =
+                        ctx.channel().attr(TIMINGS_BUILDER_KEY).get();
+                timingsBuilder.tlsHandshakeEnd();
```

Comment:
Oops, I realized that we should do this in the following locations:
```java
if (handshakeFailed) {
    // The connection will be closed automatically by SslHandler.
    timingsBuilder.tlsHandshakeEnd();
    return;
}

if (!sslHandler.handshakeFuture().isDone()) {
    // Ignore successful handshake events for other SslHandlers in the pipeline.
    // This can happen when a client connects with ssl to an encrypted proxy server,
    // and then to an encrypted backend. Without this check, clients may incorrectly
    // conclude protocol negotiation and start a new session.
    return;
}
timingsBuilder.tlsHandshakeEnd();
```

Reply:
Oh, thank you. i will fix it!!
I think this fix also solved the failure of the proxy server unit test code!